tests/components/hassio/test_backup.py

```diff
@@ -0,0 +1,246 @@
+"""Test supervisor backup functionality."""
+
+from collections.abc import AsyncGenerator, Generator
+from datetime import datetime
+from io import StringIO
+import os
+from unittest.mock import AsyncMock, patch
+
+# from aiohasupervisor import backups as supervisor_backups
+from aiohasupervisor.models import backups as supervisor_backups
+import pytest
+
+from homeassistant.components.backup import DOMAIN as BACKUP_DOMAIN, BaseBackup
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .test_init import MOCK_ENVIRON
+
+from tests.typing import ClientSessionGenerator, WebSocketGenerator
+
+TEST_BACKUP = supervisor_backups.Backup(
+    compressed=False,
+    content=supervisor_backups.BackupContent(
+        addons=["ssl"],
+        folders=["share"],
+        homeassistant=True,
+    ),
+    date=datetime.fromisoformat("1970-01-01T00:00:00Z"),
+    location="local",
+    name="Test",
+    protected=False,
+    size=1.0,
+    slug="abc123",
+    type=supervisor_backups.BackupType.PARTIAL,
+)
+
+
+@pytest.fixture(autouse=True)
+def fixture_supervisor_environ() -> Generator[None]:
+    """Mock os environ for supervisor."""
+    with patch.dict(os.environ, MOCK_ENVIRON):
+        yield
+
+
+@pytest.fixture(autouse=True)
+async def setup_integration(
+    hass: HomeAssistant, supervisor_client: AsyncMock
+) -> AsyncGenerator[None]:
+    """Set up Kitchen Sink integration."""
+    with (
+        patch("homeassistant.components.backup.is_hassio", return_value=True),
+        patch("homeassistant.components.backup.backup.is_hassio", return_value=True),
+    ):
+        assert await async_setup_component(hass, BACKUP_DOMAIN, {BACKUP_DOMAIN: {}})
```

Comment:
Where or what is setting up the hassio integration? Bootstrap isn't called for tests, as far as I know.

Reply:
It's set up by the `hassio_stubs` fixture which is used by the `hassio_client` fixture which the tests use.

Reply:
Thanks!